src/reducers/versions.spec.tsx

```diff
@@ -117,12 +120,48 @@ describe(__filename, () => {
 
       expect(state).toEqual({
         ...initialState,
+        expandedPaths: getParentFolders(version.file.selected_file),
         versionInfo: {
           [version.id]: createInternalVersion(version),
         },
       });
     });
 
+    it('prunes expandedPaths when loading version info', () => {
+      const version = { ...fakeVersion, id: nextUniqueId() };
+      const newVersion = { ...fakeVersion, id: nextUniqueId() };
+      const _getPrunedExpandedPaths = jest.fn(getPrunedExpandedPaths);
+      const _reducer = createReducer({ _getPrunedExpandedPaths });
+
+      const state = _reducer(undefined, actions.loadVersionInfo({ version }));
+      _reducer(state, actions.loadVersionInfo({ version: newVersion }));
+
+      expect(_getPrunedExpandedPaths).toHaveBeenCalledWith(
+        state.expandedPaths,
+        createInternalVersion(newVersion),
+      );
+    });
+
+    it('prunes expanded paths when setCurrentVersionId is dispatched', () => {
+      const version = { ...fakeVersion, id: nextUniqueId() };
+      const newVersion = { ...fakeVersion, id: nextUniqueId() };
+      const _getPrunedExpandedPaths = jest.fn(getPrunedExpandedPaths);
+      const _reducer = createReducer({ _getPrunedExpandedPaths });
+
+      let state = _reducer(undefined, actions.loadVersionInfo({ version }));
+      state = _reducer(state, actions.loadVersionInfo({ version: newVersion }));
+      _getPrunedExpandedPaths.mockClear();
+      _reducer(
+        state,
+        actions.setCurrentVersionId({ versionId: newVersion.id }),
+      );
+
+      expect(_getPrunedExpandedPaths).toHaveBeenCalledWith(
+        state.expandedPaths,
+        expect.objectContaining({ id: newVersion.id }),
```

Comment:
> This line should be `createInternalVersion(newVersion)` after we remove `version.expandedPaths` completely #1218 .

Actually, I think it's better as is because it does not depend on the implementation of `createInternalVersion()`.